tests/DotNetBoilerplate.Tests.Integration/BookStoreEndpointsTests.cs

```diff
@@ -0,0 +1,248 @@
+
+         //Assert
+        updateResponse.StatusCode.ShouldBe(HttpStatusCode.OK);
+
+        var updatedBookStore = await dbContext.BookStores.FirstOrDefaultAsync(x=>x.Id==bookStore.Id);
+
+        updatedBookStore.ShouldNotBeNull();
+        updatedBookStore!.Name.ShouldBe(updateRequest.Name);
+    }
+
+    [Fact]
+    public async Task GivenBookStoreDoesNotExists_UpdateBookStore_ShouldFail()
+    {
+        //Arrange
+        var updateRequest = new UpdateBookStoreEndpoint.Request
+        {
+            Name = "Updated book store",
+            Description = "Updated description"
+        };
+
+        var randomId = Guid.NewGuid();
+        //Act
+        var updateResponse = await testsFixture.Client.PutAsJsonAsync($"book-stores/{randomId}", updateRequest);
+
+         //Assert
+        updateResponse.StatusCode.ShouldBe(HttpStatusCode.BadRequest);
+    }
+
+    [Fact]
+    public async Task GivenBookStoresExists_BrowseBookStores_ShouldReturnAll()
+    {
+        //Arrange
+        var firstBookStore = new CreateBookStoreEndpoint.Request
+        {
+            Name = "BookStore",
+            Description = "Description"
+        };
+
+        var secondBookStore = new CreateBookStoreEndpoint.Request
+        {
+            Name = "BookStore2",
+            Description = "Description2"
+        };
+
+        //Act
+        var firstResponse = await testsFixture.Client.PostAsJsonAsync("book-stores", firstBookStore);
+
+        testsFixture.Client.DefaultRequestHeaders.Remove("Authorization");
+        await SignUpAndSignIn("email2@t.pl", "ttttttt2", "username2");
+
+        var secondResponse = await testsFixture.Client.PostAsJsonAsync("book-stores", secondBookStore);
+
+        //Assert
+        firstResponse.StatusCode.ShouldBe(HttpStatusCode.OK);
+        secondResponse.StatusCode.ShouldBe(HttpStatusCode.OK);
+
+        using var serviceScope = testsFixture.ServiceProvider.CreateScope();
+        var dbContext = serviceScope.ServiceProvider.GetRequiredService<DotNetBoilerplateReadDbContext>();
+
+        var bookStores = await dbContext.BookStores.ToListAsync();
+        bookStores.Count.ShouldBe(2);
+    }
+
+    [Fact]
+    public async Task GivenBookStoreExists_GetBookStoreById_ShouldSucceed()
+    {
+        //Arrange
+        var request = new CreateBookStoreEndpoint.Request
+        {
+            Name = "BookStore",
+            Description = "Description"
+        };
+
+        var response = await testsFixture.Client.PostAsJsonAsync("book-stores", request);
+
+        response.StatusCode.ShouldBe(HttpStatusCode.OK);
+
+        using var serviceScope = testsFixture.ServiceProvider.CreateScope();
+        var dbContext = serviceScope.ServiceProvider.GetRequiredService<DotNetBoilerplateReadDbContext>();
+
+        var bookStore = await dbContext.BookStores.FirstOrDefaultAsync();
+
+        //Act
+        await testsFixture.Client.GetAsync($"book-stores/{bookStore.Id}");
```

Comment:
Tutaj też, wywołany endpoint ale nic to nie wnosi - fire&forget. Zamiast używać db context pod spodem i na górze to po prostu po create, wyciągnjcie ID stworzonego zasobu, potem wywołajcie endpoint var getBookStoreByIdResponse await testsFixture.Client.GetAsync($"book-stores/{createBookStoreResult.Id}");, sprawdźcie czt jest 200 i czy sie zgadza np. nazwa z tym co podane w create request i koniec testu :)